src/Behat/Behat/Config/Handler/StrictHandler.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Behat Testwork.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Config\Handler;
+
+use Behat\Testwork\Tester\Result\Interpretation\StrictInterpretation;
+use Behat\Testwork\Tester\Result\ResultInterpreter;
+
+/**
+ * Enables stop on failure via config.
+ */
+final class StrictHandler
+{
+    /**
+     * @var ResultInterpreter
+     */
+    private $resultInterpreter;
+
+    public function __construct(ResultInterpreter $resultInterpreter)
+    {
+        $this->resultInterpreter = $resultInterpreter;
+    }
```

Comment:
Now that we only support PHP ^8.1, we should probably use typed properties / constructor promotion / readonly properties in new code? (we can't yet use readonly classes as they are 8.2 only).

```suggestion

    public function __construct(
        private readonly ResultInterpreter $resultInterpreter,
    ) {}
```

Reply:
I was wondering what was the policy on that kind of feature.

If you did drop older version of PHP, that's great, and I can of course migrate typing. You might also consider migrate all codebase to typed property using rectorphp and this kind of set: https://getrector.com/find-rule?rectorSet=core-type-declarations&activeRectorSetGroup=core (but if you consider that it might be breaking, it may be only on the next major).

Another "new" php feature I was wondering is the `array('foo')` vs `['foo']`. Behat does use `array` everywhere, but I think that the standard is more `['foo`]` now (but I kept `array` to be consistent too).